### PR TITLE
Fixes for serial timeouts.

### PR DESCRIPTION
**Pull request by BlackBird77**
The MultiID polling is now usable, but sometimes there are some hikups and the adapter restarts.

### CODE REVIEW

File: lib/master.js [hunk 1]
```diff
@@ -570,6 +573,7 @@ function Master(options, adapter) {
             }
 
             try {
+                console.log('DefaultDeviceId'+options.config.defaultDeviceId);
```
Comment (GermanBluefox) [inline lib/master.js]: Use spaces between +

File: lib/master.js [hunk 2]
```diff
@@ -354,7 +354,10 @@ function Master(options, adapter) {
         } else {
             pollDevice(devices[deviceIds[i]], err => {
                 devices[deviceIds[i]].err = err;
-                setImmediate(poll, i + 1, cb);
+                // Wait 50ms and call now!
+		setTimeout(function () {
```
Comment (GermanBluefox) [inline lib/master.js]: This should be configurable. Imho

Comment (GermanBluefox) [reply lib/master.js]: As configure. How many ms to wait.

File: main.js [hunk 3]
```diff
@@ -17,6 +17,7 @@ let adapter       = utils.Adapter({
 process.on('SIGINT', stop);
 
 adapter.on('ready', function () {
+    console.log("Modbus Adapter ready");
```
Comment (GermanBluefox) [inline main.js]: Use single quotas

**Comment by Apollon77** [comment]
@BlackBird77 Did you see the review comments from Bluefox?

**Comment by BlackBird77** [comment]
Yes i fix this the next days. And make a new PR.

Andre

Ingo Fischer <notifications@github.com> schrieb am Fr., 4. Okt. 2019, 08:59:

**Comment by Apollon77** [comment]
Just edit your fork and push then this PR gets updated automatically

**Comment by BlackBird77** [comment]
@GermanBluefox Ok i am done with the changes!

**Comment by Samuel-0-0** [comment]
have tested this fix, but also have this bug. when have input registers only(more than one slaves), it works, but failed when added holding registers.

**Comment by GermanBluefox** [comment]
Thank you.

**Pull request merged**